flake.nix

```diff
@@ -0,0 +1,112 @@
+{
```

Comment:
Add a comment to the top of this file (like shell.nix has), explaining to non-Nix people what this file even is? And maybe a URL to a tailscale/tailscale issue about this flake in particular and about why it's brittle for now?

Reply:
Oh, and make a new issue I guess. I assumed you had one already. Then reference it from the commit message too.

Reply:
Done, and done.